base/applications/notepad/text.c

```diff
@@ -63,71 +39,143 @@ BOOL IsTextNonZeroASCII(const void *pText, DWORD dwSize)
 
 ENCODING AnalyzeEncoding(const char *pBytes, DWORD dwSize)
 {
-    INT flags = IS_TEXT_UNICODE_STATISTICS;
+    INT flags = IS_TEXT_UNICODE_STATISTICS | IS_TEXT_UNICODE_REVERSE_STATISTICS;
 
     if (dwSize <= 1)
         return ENCODING_ANSI;
 
     if (IsTextNonZeroASCII(pBytes, dwSize))
-    {
         return ENCODING_ANSI;
-    }
 
     if (IsTextUnicode(pBytes, dwSize, &flags))
-    {
         return ENCODING_UTF16LE;
-    }
 
     if ((flags & IS_TEXT_UNICODE_REVERSE_MASK) && !(flags & IS_TEXT_UNICODE_ILLEGAL_CHARS))
-    {
         return ENCODING_UTF16BE;
-    }
 
     /* is it UTF-8? */
     if (MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, pBytes, dwSize, NULL, 0))
-    {
         return ENCODING_UTF8;
-    }
 
     return ENCODING_ANSI;
 }
 
+static VOID
+ReplaceNewLines(LPWSTR pszNew, DWORD cchNew, LPCWSTR pszOld, DWORD cchOld, WCHAR chTarget)
```

Comment:
This doesn't break reading and saving files with only LF, or only CR newlines?

Reply:
`pszNew` and `pszOld` are different buffers. No problem.

Reply:
Fixed.